### PR TITLE
Add ability to filter zero user balances

### PR DESCRIPTION
**Pull request by dpatsora**

### CODE REVIEW

File: app/api/v2/account/balances.rb [hunk 1]
```diff
@@ -19,8 +19,13 @@ class Balances < Grape::API
           use :pagination
         end
         get '/balances' do
-          present paginate(current_user.accounts.visible.ordered),
-                  with: Entities::Account
+          if params[:not_empty] == "true"
+            present paginate(current_user.accounts.visible.ordered.where.not(balance: 0)),
```
Comment (dpatsora) [inline app/api/v2/account/balances.rb]: where.not(balance: 0, locked: 0)?

File: app/api/v2/account/balances.rb [hunk 2]
```diff
@@ -19,8 +19,13 @@ class Balances < Grape::API
           use :pagination
         end
         get '/balances' do
-          present paginate(current_user.accounts.visible.ordered),
-                  with: Entities::Account
+          if params[:not_empty] == "true"
```
Comment (mnaichuk) [inline app/api/v2/account/balances.rb]: Add validations for this param.

File: app/api/v2/account/balances.rb (hunk 1 above)
Comment (mnaichuk) [inline app/api/v2/account/balances.rb]: Or you can write like `where('balance != 0')` but it's up to you

File: spec/api/v2/account/balances_spec.rb [hunk 3]
```diff
@@ -35,6 +35,20 @@
       end
     end
 
+    context 'not nil balances' do
```
Comment (mnaichuk) [inline spec/api/v2/account/balances_spec.rb]: non zero balances

File: spec/api/v2/account/balances_spec.rb [hunk 4]
```diff
@@ -35,6 +35,20 @@
...
+      before { api_get '/api/v2/account/balances', token: token, params: {not_empty: true} }
+
+      it { expect(response).to have_http_status 200 }
+
+      it 'returns current user balances' do
+        result = JSON.parse(response.body)
+        expect(result).to contain_exactly(
```
Comment (mnaichuk) [inline spec/api/v2/account/balances_spec.rb:61]: something wrong with your indentation

**Review by mnaichuk — CHANGES_REQUESTED** [review]

File: app/api/v2/account/balances.rb [hunk 5]
```diff
@@ -17,10 +17,19 @@ class Balances < Grape::API
             success: API::V2::Entities::Account
         params do
           use :pagination
+          optional :not_empty,
+                   type: String,
+                   values: { value: ["true", "false"], message: 'not allowed'},
```
Comment (mnaichuk) [inline app/api/v2/account/balances.rb]: value: %w(true false)
And fix error message it should be dot separated like here https://github.com/openware/peatio/pull/2444/files#diff-329848d08811e38fcdf3da1955c974ffR42

File: spec/api/v2/account/balances_spec.rb [hunk 6]
```diff
@@ -35,6 +35,20 @@
       end
     end
 
+    context 'non zero balances' do
+      before { api_get '/api/v2/account/balances', token: token, params: {not_empty: true} }
+
+      it { expect(response).to have_http_status 200 }
```
Comment (mnaichuk) [inline spec/api/v2/account/balances_spec.rb:57]: please add specs for negative scenarios

File: app/api/v2/account/balances.rb (hunk 5 above)
Comment (mnaichuk) [reply app/api/v2/account/balances.rb]: Add there is Boolean type for Grape validators

Comment (mnaichuk) [reply app/api/v2/account/balances.rb]: Also please set a default value

**Review by mnaichuk — CHANGES_REQUESTED** [review]

File: app/api/v2/account/balances.rb [hunk 7]
```diff
@@ -17,10 +17,19 @@ class Balances < Grape::API
             success: API::V2::Entities::Account
         params do
           use :pagination
+          optional :not_empty,
+                   type: String,
```
Comment (mnaichuk) [inline app/api/v2/account/balances.rb]: And then you can rewrite condition with a boolean value

**Review by mnaichuk — CHANGES_REQUESTED** [review]

File: spec/api/v2/account/balances_spec.rb [hunk 8]
```diff
@@ -35,6 +35,37 @@
...
+      it 'returns non zero balances' do
+        result = JSON.parse(response.body)
+        expect(result).to contain_exactly(
+                            { 'currency' => 'btc',  'balance' => '5.0',  'locked'  => '5.0' },
+                            { 'currency' => 'eth',  'balance' => '30.5', 'locked'  => '0.0' },
+                            )
+      end
+    end
+
+    context 'use non_empty parameter == false' do
```
Comment (mnaichuk) [inline spec/api/v2/account/balances_spec.rb]: and with non boolean value

**Review by mnaichuk — CHANGES_REQUESTED** [review]

File: app/api/v2/account/balances.rb [hunk 9]
```diff
@@ -17,10 +17,19 @@ class Balances < Grape::API
             success: API::V2::Entities::Account
         params do
           use :pagination
+          optional :not_empty,
```
Comment (mnaichuk) [inline app/api/v2/account/balances.rb]: please use another name

**Review by mnaichuk — CHANGES_REQUESTED** [review]

File: app/api/v2/account/balances.rb [hunk 10]
```diff
@@ -17,10 +17,19 @@ class Balances < Grape::API
             success: API::V2::Entities::Account
         params do
           use :pagination
+          optional :nonzero,
+                   type: { value: Boolean, message: 'account.balances.non_expected_value' },
```
Comment (mnaichuk) [inline app/api/v2/account/balances.rb]: invalid_nonzero

Comment (mnaichuk) [inline app/api/v2/account/balances.rb]: invalid_nonzero

**Review by mnaichuk — CHANGES_REQUESTED** [review]

**Review by ysv — APPROVED** [review]

**Review by mnaichuk — APPROVED** [review]

**Pull request merged**